security-framework/src/secure_transport.rs

```diff
@@ -1292,7 +1296,7 @@ mod test {
         p!(ctx.set_peer_domain_name("google.com"));
         let stream = p!(TcpStream::connect("google.com:443"));
         let mut stream = p!(ctx.handshake(stream));
-        p!(stream.write_all(b"GET / HTTP/1.0\r\n\r\n"));
+        p!(stream.write_all(b"GET / HTTP/1.1\r\nHost: google.com\r\n\r\n"));
```

Comment:
By setting the version to 1.1, Google should keep the connection alive, and reading to the end (without the buffered check) should then hang. Cannot confirm test on my equipment.